opm/simulators/wells/WellInterface_impl.hpp

```diff
@@ -107,6 +107,7 @@ namespace Opm
         well_efficiency_factor_ = 1.0;
 
         connectionRates_.resize(number_of_perforations_);
+        perf_densities_.resize(number_of_perforations_);
```

Comment:
This should not be used in MSW at all. So it should not go to WellInterface.

Reply:
I don't see why. The perf_densities are well defined also for MSW and computed in computePerfCellPressDiffs.

Reply:
But it is only temporary in that function, it is used for calculation of something else, then its life is finished there.  So it does not need to be a member at all. 

And also, they have different definition and different way of calculation. 

Unless there are some other evidence to extend its lifespan, we should not do that.

Reply:
Currently, it is used in calculateBhpFromThp. If you are so strongly against this, I can revert these changes and move back the cacluateBhpFromThp code. I just tried to remove some code duplication between the standard and MSW code. Any other things you want me to change in the PR?

Reply:
Please do that. Because we should only introduce working code, remove code not working.  It is not good to introduce code not working. Depending on the plan, it might take some time before we get a chance to re-factor or correct code that not working. 

Please also remove the code (not working) in the function `computeWellPotentialWithTHP()`, instead, you can just throw a warning and return some trivial values, depending on how you want to organize the `computeWellPotentials`.  You can decide whether you want to put `computeWellPotentialWithTHP` to `StandardWell` or keep it in `WellInterface`. 

Those are the big parts with a quick review.  After that, I will provide more detailed review comments. It will be easier when we reduce the code change first.